### DISCUSSION TITLE
Error message vpcPlot :

### DISCUSSION THREAD

**Discussion by Percival24** [nlmixr2/nlmixr2#132] [Q&A] [answered]
Hello, 

I am facing an error message when I try to use vpcPlot on my output of nlmixr2.  Here is what it says : 

_Error in sim$sim[sim[[id]] == i] <- cumsum(stats::ave(1:length(sim_id[sim[[id]] ==  :
  NAs are not allowed in subscripted assignments
In addition: Warning message:
In as.num(sim[[id]]) : NAs introduced by coercion_

I wonder if it is a kind of "classic error" message for vpcPlot or if someone has faced it before... 

Best regards, 
Percival

**Comment by mattfidler** [comment]
Hi @Percival24 

Not a classic error message for `vpcPlot`.  My guess it that the package `vpc` did not get what it expected from the output.

You can try to debug it yourself by using `vpcSim()` instead and then directly trying `vpc` from the `vpc` package.  There is also the tinyvpc package that can likley use the `vpcSim()` output.

**Reply by mattfidler** [reply]
You can also update to the new `nlmixr2est` just released today, it does have some release notes that this was improved.

You may have to wait a week or so if you want CRAN binaries as well.

**Reply by Percival24** [reply, marked as answer]
The problem has been resolved by changing the nature of my variable ID. 
The ID variable is the variable that identifies each subject by a reference number. 
What I did is defining that reference number as an integer instead of a character. 
That worked. 
But thanks again for your reply !!!!! :)
